### DISCUSSION TITLE
Is there a way to change the backdrop root element?

### DISCUSSION THREAD

**Discussion by gsartori** [twbs/bootstrap#39302] [Q&A]
We need the backdrop of a modal/offcanvas to cover only a specific area of our GUI, is there a way of configuring a root element for the backdrop? Something different from `<body>`?

I'm actually detaching/appending the backdrop `onShown` but there is a flicker of course.

Any hint/idea would be really appreciated,
Thank you,
Gianluca

**Comment by julien-deramond** [comment]
Hey @gsartori
To my knowledge, there's no interface in the JS to do that (with a `data-bs-something` for example) and define a specific target in your DOM as a container for your backdrop. 
As you can see in the following, `rootElement` is in the Offcanvas component black box and is not directly accessible.

```js
return new Backdrop({
  className: CLASS_NAME_BACKDROP,
  isVisible,
  isAnimated: true,
  rootElement: this._element.parentNode,
  clickCallback: isVisible ? clickCallback : null
})
```

Depending on what you're trying to achieve, you could maybe use the [static backdrop](https://getbootstrap.com/docs/5.3/components/offcanvas/#static-backdrop) and the modify the CSS rendering of `.offcanvas-backdrop` 🤷
